xonsh/completers/completer.py

```diff
@@ -125,3 +190,41 @@ def remove_completer(name: str):
         return
     else:
         return None, err + "\n", 1
+
+
+class CompleterAlias(ArgParserAlias):
+    """CLI to add/remove/list xonsh auto-complete functions"""
+
+    def build(self):
+        parser = self.create_parser(prog="completer")
+        parser.add_command(_register_completer, prog="add")
+        parser.add_command(remove_completer, prog="remove", aliases=["rm"])
+        parser.add_command(list_completers, prog="list", aliases=["ls"])
+        return parser
+
+
+completer_alias = CompleterAlias()
+
+
+@contextual_command_completer
+def complete_argparser_aliases(command: CommandContext):
+    """
+    Completion for "completer"
+    """
+    if not command.args:
+        return
+    cmd = command.args[0].value
+
+    alias = XSH.aliases.get(cmd)  # type: ignore
+    if not hasattr(alias, "parser"):
```

Comment:
previously I had that version. But the isinstance thing failed, because of different version from amalgamated class instance and normal import paths. This way one can just set the attribute as well.

Reply:
I see. Can you add a todo comment above? I intend on fixing this amalgamation issue one day

Reply:
sure will do. I plan to create a new PR just for ArgParserAlias and completer alias though. I will put the changes there. So you will not be having the noise from changes for other aliases